src/labels.rs

```diff
@@ -18,61 +21,82 @@ pub const APP_PART_OF_LABEL: &str = concatcp!(APP_KUBERNETES_LABEL_BASE, "part-o
 pub const APP_MANAGED_BY_LABEL: &str = concatcp!(APP_KUBERNETES_LABEL_BASE, "managed-by");
 pub const APP_ROLE_GROUP_LABEL: &str = concatcp!(APP_KUBERNETES_LABEL_BASE, "role-group");
 
+/// Recommended labels to set on objects created by Stackable operators
+///
+/// See [`get_recommended_labels`] and [`ObjectMetaBuilder::with_recommended_labels`].
+#[derive(Debug, Clone, Copy)]
+pub struct ObjectLabels<'a, T> {
+    /// The name of the object that this object is being created on behalf of, such as a `ZookeeperCluster`
+    pub owner: &'a T,
+    /// The name of the app being managed, such as `zookeeper`
+    pub app_name: &'a str,
+    /// The version of the app being managed (not of the operator)
+    pub app_version: &'a str,
```

Comment:
Could we please document what exactly is meant to be put in here? is it
* 3.3.1
* 3.3.1-stackable0.1.0
* One of the above
* It doesn't really matter, its for documentation purpose

AFAIK we currently put in `3.3.1-stackable0.1.0` but will need to switch to `3.3.1` when implementing product image select (as we don't know the stackable version in all cases)

Reply:
I'd say a combination of 1 and 4. But sure.

Reply:
Actually, on second thought some properties depend on the stackable release too so maybe it just makes sense to put in both...

Reply:
caf2ac6

Reply:
I would prefer `3.3.1-stackable0.1.0`, but we don't have the information of the stackable version `0.1.0` if the users provides a custom image.
Is it ok in that case to put in `3.3.1`? If so, would be great to have this documented to avoid further discussion.
E.g.

This version should include the Stackable version, such as `3.0.0-stackable0.1.0`. If the Stackable version is not known, only the product version can be used, such as `3.0.0`. However, this is pure documentation and should not be parsed.

Reply:
> but we don't have the information of the stackable version 0.1.0 if the users provides a custom image.

In that case I'd just put whatever tag the user specified. Or `<version>-<tag>` or some such.

Reply:
(Hopefully whatever tagging system the user chooses to use actually makes sense to them, at least...)

Reply:
Ah that's a good idea. So new suggestion

This version should include the Stackable version, such as 3.0.0-stackable0.1.0. If the Stackable version is not known, the product version should be used together with a suffix (if possible). In the case of custom product images provided by the user, only the product version is known, `3.0.0-<tag-of-custom-image>` should be used. However, this is pure documentation and should not be parsed.

Reply:
165c1943511220a2bf81836a278c41199cdf7f65

Reply:
thx!